### PR TITLE
Fix comprehensive txtfile bugs and add thorough tests for repeat date

### PR DESCRIPTION
**Pull request by dsicol**

### CODE REVIEW

File: src/main/java/seedu/exceptions/ExceptionChecker.java [hunk 1]
```diff
@@ -23,7 +23,8 @@ public static void checkValidAmount(double amountVal) throws NotPositiveValueExc
 
     public static void checkValidDoubleInput(String amountVal) throws InvalidCharacterInAmount {
         String lowerCaseAmountVal = amountVal.toLowerCase();
-        boolean containsSpecialCharacter = (lowerCaseAmountVal.contains("f") || lowerCaseAmountVal.contains("d"));
+        boolean containsSpecialCharacter = (lowerCaseAmountVal.contains("f") || lowerCaseAmountVal.contains("d") ||
+                lowerCaseAmountVal.contains("e"));
```
Comment (itszhixuan) [inline src/main/java/seedu/exceptions/ExceptionChecker.java:27]: Good addition of catching 'e'

**Review by itszhixuan — APPROVED** [review]
Protects txt file from being edited. Good coverage

File: src/main/java/seedu/exceptions/TxtFileException.java
Comment (Chick3nBoy) [inline src/main/java/seedu/exceptions/TxtFileException.java:1]: Nice work creating a new exception for the specific case.

File: src/main/java/seedu/txtdata/TxtFileStatus.java [hunk 2]
```diff
@@ -80,47 +82,96 @@ public static void saveExpenditureList(ArrayList<Expenditure> expenditures) thro
...
+                    break;
+                case "Tu":
+                    initializeTuitionExpenditure(saveData, expenditures);
+                    break;
+                default:
+                    throw new TxtFileException();
+                }
+            } catch (TxtFileException t) {
+                System.out.println(t.getMessage());
+            } catch (DateTimeParseException | NumberFormatException | ArrayIndexOutOfBoundsException e) {
```
Comment (Chick3nBoy) [inline src/main/java/seedu/txtdata/TxtFileStatus.java:130]: Good working catching the exception in txtFileStatus instead for a more compact MyLedger.

**Review by Chick3nBoy — APPROVED** [review]

**Pull request merged**